Sources/ServiceLifecycle/ServiceGroup.swift

```diff
@@ -584,6 +650,43 @@ public actor ServiceGroup: Sendable {
             throw error
         }
     }
+
+    private func cancelGroupAndSpawnTimeoutIfNeeded(
+        group: inout ThrowingTaskGroup<ChildTaskResult, Error>,
+        cancellationTimeoutTask: inout Task<Void, Never>?
+    ) {
+        guard cancellationTimeoutTask == nil else {
+            // We already have a cancellation timeout task running.
+            return
+        }
+        group.cancelAll()
+
+
+        if #available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *), let maximumCancellationDuration = self.maximumCancellationDuration {
+            // We have to spawn an unstructured task here because the call to our `run`
+            // method might have already been cancelled and we need to protect the sleep
+            // from being cancelled.
+            cancellationTimeoutTask = Task {
+                do {
+                    self.logger.debug(
+                        "Task cancellation timeout task started."
+                    )
+                    try await Task.sleep(for: Duration(
+                        secondsComponent: maximumCancellationDuration.secondsComponent,
+                        attosecondsComponent: maximumCancellationDuration.attosecondsComponent
+                    ))
```

Comment:
`maximumCancellationDuration` is already a `Duration` here, I think?

Reply:
No it's not. We are storing the raw components again because we cannot store a `Duration` due to the availability